src/web/core/src/main/java/org/geoserver/web/data/store/DefaultDataStoreEditPanel.java

```diff
@@ -160,13 +160,15 @@ private Panel getInputComponent(final String componentId, final IModel paramsMod
         final Class<?> binding = paramMetadata.getBinding();
         final List<Serializable> options = paramMetadata.getOptions();
 
+        boolean visible = true;
+
         Panel parameterPanel;
         if("dbtype".equals(paramName) || "filetype".equals(paramName)) {
```

Comment:
I've attempted an approach using the GeoTools parameters, See https://github.com/geotools/geotools/pull/1659 and https://github.com/geoserver/geoserver/pull/2472